include/mra/misc/convolutiondata.h

```diff
@@ -204,6 +229,70 @@ namespace mra {
       }
     };
 
+  template <typename T, Dimension NDIM>
+  class Operator {
+
+  private:
+    size_type K;
+    std::map<Key<NDIM>, OperatorData<T, NDIM>> opdata;     // map for storing operator data
+    std::mutex cachemutex;                                 // mutex for thread safety
+
+    T norm_ns(Level n, ConvolutionData<T>* const ns[]) const {
+      T norm = 1.0, sum = 0.0;
+
+      for (size_type d = 0; d < NDIM; ++d) {
+        TensorView<T, 2> ns_rview(ns[d]->R);
```

Comment:
The right thing here is to create a new tensor, not to play games with constness.